### PR TITLE
Use hf filesystem for loading component

### PR DESCRIPTION
**Pull request by NielsRogge**
This PR uses the new `hf://` fsspec specification to read a Dask dataframe directly from the hub.

### CODE REVIEW

File: examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py [hunk 1]
```diff
@@ -46,9 +44,7 @@ def load(self, args):
         """
         # 1) Load data, read as Dask dataframe
         logger.info("Loading dataset from the hub...")
-        dataset = load_dataset(args.dataset_name, split="train")
-        dataset.to_parquet("data.parquet")
-        dask_df = dd.read_parquet("data.parquet")
+        dask_df = dd.read_parquet("hf://datasets/lambdalabs/pokemon-blip-captions/data/train-00000-of-00001-566cc9b19d7203f8.parquet")
```
Comment (RobbeSneyders) [inline examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py:47]: We still want to receive the dataset as an argument, right?

File: examples/pipelines/simple_pipeline/components/load_from_hub/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-datasets==2.11.0
+huggingface_hub==0.14.1
```
Comment (RobbeSneyders) [inline examples/pipelines/simple_pipeline/components/load_from_hub/requirements.txt:1]: Do we need this? I don't see this used explicitly in the code.

Comment (NielsRogge) [reply examples/pipelines/simple_pipeline/components/load_from_hub/requirements.txt:1]: This is needed for the HF filesystem

File: examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py (hunk 1 above)
Comment (NielsRogge) [reply examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py]: Ok, fixed that

File: examples/pipelines/simple_pipeline/components/load_from_hub/requirements.txt (hunk 2 above)
Comment (RobbeSneyders) [reply examples/pipelines/simple_pipeline/components/load_from_hub/requirements.txt:1]: Yes but how is it used if we are not calling it directly? Is it a subdependency of another package?

File: examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py [hunk 3]
```diff
@@ -46,9 +44,7 @@ def load(self, args):
         """
         # 1) Load data, read as Dask dataframe
         logger.info("Loading dataset from the hub...")
-        dataset = load_dataset(args.dataset_name, split="train")
-        dataset.to_parquet("data.parquet")
-        dask_df = dd.read_parquet("data.parquet")
+        dask_df = dd.read_parquet(f"hf://datasets/{args.dataset_name}")
```
Comment (RobbeSneyders) [inline examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py:47]: What is the format of the argument we're expecting here now?

Comment (NielsRogge) [reply examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py:47]: The name of a dataset on the HF hub, like `lambdalabs/pokemon-blip-captions`.

File: fondant/dataset.py [hunk 4]
```diff
@@ -113,11 +113,13 @@ def _upload_index(self, df: dd.DataFrame):
 
     def _upload_subset(self, name: str, fields: Mapping[str, Field], df: dd.DataFrame):
         # add subset to the manifest
-        manifest_fields = [(field.name, Type[field.type]) for field in fields.values()]
+        manifest_fields = [
+            (field.name, Type[field.type.name]) for field in fields.values()
+        ]
```
Comment (NielsRogge) [inline fondant/dataset.py:118]: Had to include this fix to make the component run successfully

**Comment by NielsRogge** [comment]
@RobbeSneyders feel free to approve

**Comment by RobbeSneyders** [comment]
Can you still respond to my open question on the dependency?

File: examples/pipelines/simple_pipeline/components/load_from_hub/requirements.txt (hunk 2 above)
Comment (NielsRogge) [reply examples/pipelines/simple_pipeline/components/load_from_hub/requirements.txt:1]: It's a subdependency of `fsspec`, see https://huggingface.co/docs/huggingface_hub/main/package_reference/hf_file_system

File: fondant/dataset.py (hunk 4 above)
Comment (PhilippeMoussalli) [reply fondant/dataset.py:118]: Had the same issue in my PR

**Review by RobbeSneyders — APPROVED** [review]
Thanks @NielsRogge!

**Pull request merged**